src/painting/painter.rs

```diff
@@ -243,6 +243,9 @@ impl Painter {
 
         // Moving the start position of the cursor based on the size of the required lines
         if self.large_buffer || is_reset() {
+            for _ in 0..screen_height - lines.required_lines(screen_width, None) {
+                self.stdout.queue(Print(&coerce_crlf("\n")))?;
+            }
```

Comment:
@qilme @Bahex Sorry to step in, I've noticed a panic caused by this change.

Basically we need `saturating_sub` to avoid negative numbers.
Besides, I'm not sure whether we should take `lines.hint` into account here.

Reply:
> getting a multiline hint can similarly push text up, not taking that into account can cause lines to disappear as well.

Say we have a large_buffer repaint caused by a long multiline hint, if we add the height of self.hint to `lines.required_lines`, we'll end up not doing any scrolling. Is that expected behavior?